<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Hao He" />

<meta name="date" content="2020-02-16" />

<title>plot_conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->





<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CSNA_workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/TheJacksonLaboratory/CSNA">GitHub</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/CSNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">plot_conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019</h1>
<h4 class="author"><em>Hao He</em></h4>
<h4 class="date"><em>2020-02-16</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#plot-scan-result-for-conditional_m2.novelty_residuals_ranknormal_datarelease_12182019">plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-02-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>csna_workflow/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190918code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190918)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190918code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190918)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryCSNAtreee7e3e2c40ccf302494fa1a8d920936b31b61ec55targetblanke7e3e2ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/TheJacksonLaboratory/CSNA/tree/e7e3e2c40ccf302494fa1a8d920936b31b61ec55" target="_blank">e7e3e2c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryCSNAtreee7e3e2c40ccf302494fa1a8d920936b31b61ec55targetblanke7e3e2ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/.nfs0000000135a08b9b00002448
    Untracked:  analysis/.nfs0000000135a08ba50000244a
    Untracked:  analysis/.nfs0000000159105b68000007b2
    Untracked:  analysis/01_geneseek2qtl2.script
    Untracked:  analysis/02_geneseek2intensity.script
    Untracked:  analysis/03_2_firstgm2genoprobs.R
    Untracked:  analysis/03_2_firstgm2genoprobs.script
    Untracked:  analysis/03_firstgm2genoprobs.script
    Untracked:  analysis/04_diagnosis_qc_gigamuga_11_batches.R
    Untracked:  analysis/04_diagnosis_qc_gigamuga_11_batches.script
    Untracked:  analysis/04_diagnosis_qc_gigamuga_nine_batches.script
    Untracked:  analysis/05_after_diagnosis_qc_gigamuga_11_batches.R
    Untracked:  analysis/05_after_diagnosis_qc_gigamuga_11_batches.script
    Untracked:  analysis/05_after_diagnosis_qc_gigamuga_nine_batches.script
    Untracked:  analysis/06_final_pr_apr_69K.script
    Untracked:  analysis/07.1_html_founder_prop.script
    Untracked:  analysis/07_recomb_size_founder_prop.script
    Untracked:  analysis/08.0_11_batches_gcta.sh
    Untracked:  analysis/08.1_11_batches_merge_plink.sh
    Untracked:  analysis/08.1_11_batches_merge_plink.sh.e11230968
    Untracked:  analysis/08.1_11_batches_merge_plink.sh.o11230968
    Untracked:  analysis/12_plot_qtl_mapping.R
    Untracked:  analysis/12_plot_qtl_mapping.script
    Untracked:  analysis/13_conditional_blup.R
    Untracked:  analysis/13_conditional_m2_qtlmapping.R
    Untracked:  analysis/13_conditional_m2_qtlmapping.script
    Untracked:  analysis/13_conditional_qtlmapping.R
    Untracked:  analysis/13_plot_conditional_m2_qtlmapping.R
    Untracked:  analysis/13_plot_conditional_m2_qtlmapping.script
    Untracked:  analysis/14_geneseek2qtl2fst_Gagnon.R
    Untracked:  analysis/14_geneseek2qtl2fst_Gagnon.script
    Untracked:  analysis/15_firstgm2genoprobs_Gagnon.R
    Untracked:  analysis/15_firstgm2genoprobs_Gagnon.script
    Untracked:  analysis/16_diagnosis_qc_gigamuga_gagnon.R
    Untracked:  analysis/16_diagnosis_qc_gigamuga_gagnon.script
    Untracked:  analysis/17_qtlmapping.R
    Untracked:  analysis/Novelty_no_outliers_datarelease_12182019.gcta.script
    Untracked:  analysis/Novelty_raw_datarelease_12182019.gcta.script
    Untracked:  analysis/Novelty_resids_datarelease_12182019.gcta.script
    Untracked:  analysis/Novelty_resids_datarelease_12182019_m1.script
    Untracked:  analysis/Novelty_resids_datarelease_12182019_m2.script
    Untracked:  analysis/Novelty_resids_datarelease_12182019_m3.script
    Untracked:  analysis/Novelty_residuals_RankNormal_datarelease_12182019.gcta.script
    Untracked:  analysis/Novelty_residuals_RankNormal_datarelease_12182019_m1.script
    Untracked:  analysis/Novelty_residuals_RankNormal_datarelease_12182019_m2.script
    Untracked:  analysis/Novelty_residuals_RankNormal_datarelease_12182019_m3.script
    Untracked:  analysis/Prj01_RL-Acquisition_preqc_02142020_m2.script
    Untracked:  analysis/Prj01_RL-Reversal_preqc_02142020_m2.script
    Untracked:  analysis/Prj02_Sensitization_preqc_02142020_m2.script
    Untracked:  analysis/Prj04_IVSA_preqc_02142020_m2.script
    Untracked:  analysis/blup_14_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_14_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_14_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_14_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_14_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_14_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_15_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_15_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_15_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_15_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_15_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_15_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_16_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_16_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_16_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_16_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_16_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_16_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_17_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_17_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_17_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_17_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_17_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_17_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_18_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_18_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_18_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_18_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_18_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_18_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_19_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_19_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_19_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_19_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_19_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_19_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_20_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_20_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_20_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_20_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_20_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_20_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_21_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_21_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_21_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_21_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_21_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_21_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_22_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_22_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_22_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_22_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_22_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_22_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_23_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_23_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_23_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_23_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_23_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_23_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_24_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_24_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_24_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_24_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_24_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_24_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_25_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_25_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_25_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_25_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_25_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_25_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_26_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_26_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_26_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_26_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_26_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_26_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_27_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_27_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_27_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_27_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_27_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_27_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_28_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_28_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_28_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_28_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_28_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_28_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_29_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_29_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_29_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_29_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_29_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_29_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_30_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_30_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_30_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_30_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_30_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_30_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_31_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_31_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_31_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_31_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_31_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_31_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_32_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_32_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_32_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_32_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_32_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_32_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_33_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_33_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_33_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_33_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_33_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_33_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_34_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_34_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_34_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_34_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_34_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_34_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_35_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_35_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_35_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_35_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_35_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_35_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_36_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_36_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_36_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_36_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_36_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_36_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_37_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_37_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_37_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_37_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_37_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_37_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_38_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_38_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_38_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_38_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_38_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_38_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_39_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_39_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_39_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_39_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_39_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_39_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_40_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_40_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_40_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_40_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_40_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_40_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_41_m1_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_41_m1_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_41_m2_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_41_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/blup_41_m3_Novelty_resids_datarelease_12182019.script
    Untracked:  analysis/blup_41_m3_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_Novelty_residuals_RankNormal_datarelease_12182019_m2.script
    Untracked:  analysis/conditional_blup_14_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_15_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_16_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_17_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_18_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_19_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_20_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_21_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_22_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_23_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_24_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_25_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_26_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_27_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_28_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_29_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_30_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_31_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_32_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_33_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_34_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_35_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_36_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_37_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_38_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_39_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_40_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/conditional_blup_41_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_10_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_11_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_12_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_13_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_14_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_15_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_16_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_17_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_18_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_19_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_1_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_20_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_21_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_22_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_23_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_24_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_25_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_26_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_27_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_28_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_29_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_2_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_30_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_31_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_32_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_33_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_34_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_35_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_36_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_37_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_38_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_39_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_3_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_40_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_41_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_42_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_43_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_44_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_45_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_46_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_47_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_48_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_49_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_4_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_50_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_5_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_6_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_7_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_8_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/permu_9_m2_Novelty_residuals_RankNormal_datarelease_12182019.script
    Untracked:  analysis/run_04_diagnosis_qc_gigamuga_11_batches.R
    Untracked:  analysis/run_05_after_diagnosis_qc_gigamuga_11_batches.R
    Untracked:  analysis/run_10_qtl_permu.o11472940
    Untracked:  analysis/run_12_plot_qtl_mapping.R
    Untracked:  analysis/run_13_conditional_blup.R
    Untracked:  analysis/run_13_conditional_qtlmapping.R
    Untracked:  analysis/run_13_conditional_qtlmapping.R.e11509917
    Untracked:  analysis/run_13_conditional_qtlmapping.R.o11509917
    Untracked:  analysis/run_13_plot_conditional_m2_qtlmapping.R
    Untracked:  analysis/run_14_geneseek2qtl2fst_Gagnon.R
    Untracked:  analysis/run_15_firstgm2genoprobs_Gagnon.R
    Untracked:  analysis/run_16_diagnosis_qc_gigamuga_gagnon.R
    Untracked:  analysis/run_17_qtlmapping.R
    Untracked:  analysis/run_run_09_qtlmapping.script
    Untracked:  analysis/run_run_10_qtl_permu.script
    Untracked:  analysis/run_run_11_qtl_blup.script
    Untracked:  analysis/run_run_13_conditional_blup.script
    Untracked:  analysis/scriptout/
    Untracked:  code/reconst_utils.R
    Untracked:  data/FinalReport/
    Untracked:  data/GCTA/
    Untracked:  data/GM/
    Untracked:  data/Jackson_Lab_11_batches/
    Untracked:  data/Jackson_Lab_Bubier_MURGIGV01/
    Untracked:  data/Jackson_Lab_Gagnon/
    Untracked:  data/cc_variants.sqlite
    Untracked:  data/marker_grid_0.02cM_plus.txt
    Untracked:  data/mouse_genes_mgi.sqlite
    Untracked:  data/pheno/
    Untracked:  output/AfterQC_Percent_missing_genotype_data.pdf
    Untracked:  output/AfterQC_Proportion_matching_genotypes_after_removal_of_bad_samples.pdf
    Untracked:  output/AfterQC_Proportion_matching_genotypes_before_removal_of_bad_samples.pdf
    Untracked:  output/AfterQC_number_crossover.pdf
    Untracked:  output/Histgram_Prj01_RL-Acquisition_preqc_02142020.pdf
    Untracked:  output/Histgram_Prj01_RL-Reversal_preqc_02142020.pdf
    Untracked:  output/Histgram_Prj02_Sensitization_preqc_02142020.pdf
    Untracked:  output/Histgram_Prj04_IVSA_preqc_02142020.pdf
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted.pdf
    Untracked:  output/Percent_missing_genotype_data.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker.pdf
    Untracked:  output/Proportion_matching_genotypes_after_removal_of_bad_samples.pdf
    Untracked:  output/Proportion_matching_genotypes_before_removal_of_bad_samples.pdf
    Untracked:  output/blup/
    Untracked:  output/condi.m2.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/condi.m2.qtl.blup_Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/condi.m2.qtl.out.RData
    Untracked:  output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/conditional.m2.qtl.out.RData
    Untracked:  output/conditional_m2.Novelty_residuals_RankNormal_datarelease_07302918_genomescan.pdf
    Untracked:  output/conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/genotype_error_marker.pdf
    Untracked:  output/genotype_frequency_marker.pdf
    Untracked:  output/m1.Novelty_resids_datarelease_07302918.RData
    Untracked:  output/m1.Novelty_resids_datarelease_07302918_coeffgeneplot.pdf
    Untracked:  output/m1.Novelty_resids_datarelease_07302918_genomescan.pdf
    Untracked:  output/m1.Novelty_resids_datarelease_12182019.RData
    Untracked:  output/m1.Novelty_resids_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m1.Novelty_resids_datarelease_12182019_genomescan.pdf
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_07302918.RData
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_07302918_genomescan.pdf
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/m2.Novelty_resids_datarelease_07302918.RData
    Untracked:  output/m2.Novelty_resids_datarelease_07302918_genomescan.pdf
    Untracked:  output/m2.Novelty_resids_datarelease_12182019.RData
    Untracked:  output/m2.Novelty_resids_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m2.Novelty_resids_datarelease_12182019_genomescan.pdf
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_07302918.RData
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_07302918_genomescan.pdf
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/m2.Prj02_Sensitization_preqc_02142020.RData
    Untracked:  output/m3.Novelty_resids_datarelease_07302918.RData
    Untracked:  output/m3.Novelty_resids_datarelease_07302918_genomescan.pdf
    Untracked:  output/m3.Novelty_resids_datarelease_12182019.RData
    Untracked:  output/m3.Novelty_resids_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m3.Novelty_resids_datarelease_12182019_genomescan.pdf
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_07302918.RData
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_07302918_genomescan.pdf
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/num.geno.pheno.in.Novelty_resids_datarelease_07302918.csv
    Untracked:  output/num.geno.pheno.in.Novelty_resids_datarelease_12182019.csv
    Untracked:  output/num.geno.pheno.in.Novelty_residuals_RankNormal_datarelease_07302918.csv
    Untracked:  output/num.geno.pheno.in.Novelty_residuals_RankNormal_datarelease_12182019.csv
    Untracked:  output/number_crossover.pdf
    Untracked:  output/permu/
    Untracked:  output/prop_across_generation_chr_p.RData

Unstaged changes:
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/01_geneseek2qtl2.R
    Modified:   analysis/02_geneseek2intensity.R
    Modified:   analysis/03_firstgm2genoprobs.R
    Modified:   analysis/04_diagnosis_qc_gigamuga_nine_batches.R
    Modified:   analysis/06_final_pr_apr_69K.R
    Modified:   analysis/07.1_html_founder_prop.R
    Modified:   analysis/07_recomb_size_founder_prop.R
    Modified:   analysis/08_gcta_herit.R
    Modified:   analysis/09_qtlmapping.R
    Modified:   analysis/10_qtl_permu.R
    Modified:   analysis/11_qtl_blup.R
    Modified:   analysis/run_01_geneseek2qtl2.R
    Modified:   analysis/run_02_geneseek2intensity.R
    Modified:   analysis/run_03_firstgm2genoprobs.R
    Modified:   analysis/run_04_diagnosis_qc_gigamuga_nine_batches.R
    Modified:   analysis/run_05_after_diagnosis_qc_gigamuga_nine_batches.R
    Modified:   analysis/run_06_final_pr_apr_69K.R
    Modified:   analysis/run_07.1_html_founder_prop.R
    Modified:   analysis/run_07_recomb_size_founder_prop.R
    Modified:   analysis/run_08_gcta_herit.R
    Modified:   analysis/run_09_qtlmapping.R
    Modified:   analysis/run_10_qtl_permu.R
    Modified:   analysis/run_11_qtl_blup.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/a2f202a3bf3f23ec2323e324d0b64fbbaccecae0/docs/13_plot_conditional_m2_qtlmapping.html" target="_blank">a2f202a</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/756b79a54cda14d350bbc55a47e08db3a3c47b09/analysis/13_plot_conditional_m2_qtlmapping.Rmd" target="_blank">756b79a</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
<td>
13_plot_conditional_m2_qtlmapping.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/13_plot_conditional_m2_qtlmapping.html" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/0294e575e3193a19c35e7b67f287caf66b54a232/analysis/13_plot_conditional_m2_qtlmapping.Rmd" target="_blank">0294e57</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
<td>
13_plot_conditional_m2_qtlmapping.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/626668d6004cd9658e35319050fa2fa8d28b608e/docs/13_plot_conditional_m2_qtlmapping.html" target="_blank">626668d</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/13_plot_conditional_m2_qtlmapping.html" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/ae79216b0d138fb48e3b5c98ad1913d58f3eca72/analysis/13_plot_conditional_m2_qtlmapping.Rmd" target="_blank">ae79216</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
<td>
13_plot_conditional_m2_qtlmapping.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/efed06b90051d37b076aa4728682439356b6d8c7/docs/13_plot_conditional_m2_qtlmapping.html" target="_blank">efed06b</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/4472c2a5296d0c50b695d90ab955ac7ac80d3956/analysis/13_plot_conditional_m2_qtlmapping.Rmd" target="_blank">4472c2a</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-21
</td>
<td>
13_plot_conditional_m2_qtlmapping.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/f187b9966748583771866baa95e9cd4340805056/docs/13_plot_conditional_m2_qtlmapping.html" target="_blank">f187b99</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/613db85bba45053ad4b0a96b8166753626ead2f2/analysis/13_plot_conditional_m2_qtlmapping.Rmd" target="_blank">613db85</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-21
</td>
<td>
13_plot_conditional_m2_qtlmapping.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/d25bab6450bafb744e64c878fa0311710260a1f8/docs/13_plot_conditional_m2_qtlmapping.html" target="_blank">d25bab6</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/45cb7a724673a3e314e93d818f62e9c95dfd02f4/analysis/13_plot_conditional_m2_qtlmapping.Rmd" target="_blank">45cb7a7</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-21
</td>
<td>
13_plot_conditional_m2_qtlmapping.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="plot-scan-result-for-conditional_m2.novelty_residuals_ranknormal_datarelease_12182019" class="section level1">
<h1>plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019</h1>
<pre class="r"><code>#   This script performs plotting on qtl conditional mapping for DO mice m2, Novelty_residuals_RankNormal_datarelease_12182019
library(qtl2)
library(ggplot2)
load(&quot;data/Jackson_Lab_11_batches/gm_DO2816_qc.RData&quot;)
pheno &lt;- read.csv(&quot;data/pheno/Novelty_residuals_RankNormal_datarelease_12182019.csv&quot;, header = TRUE)

print(&quot;m2&quot;)</code></pre>
<pre><code>[1] &quot;m2&quot;</code></pre>
<pre class="r"><code>load(&quot;output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019.RData&quot;)

#permutation results
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.
operm &lt;- list()
for(i in c(1:50)){
  operm[[i]] &lt;- get(load(paste0(&quot;output/permu/m2_Novelty_residuals_RankNormal_datarelease_12182019_&quot;, colnames(pheno)[14], &quot;_&quot;, i, &quot;.RData&quot;)))
  operm[[i]] &lt;- operm[[i]]$OFA.total.distance.traveled
}
operm &lt;- matrix(do.call(rbind, operm[c(1:50)]),ncol = 1)
class(operm) &lt;- c(&quot;scan1perm&quot;, &quot;matrix&quot;)
cutoff &lt;- summary(operm, c(0.05, 0.1))[,1]

#genome-wide results
pdf(file = paste0(&quot;output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019&quot;, &quot;_genomescan.pdf&quot;), width = 14)
for(i in names(condi.m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx &lt;- maxlod(condi.m2.qtl.out[[i]]) # overall maximum LOD score
  plot(condi.m2.qtl.out[[i]], gm_DO2816_qc$gmap, lodcolumn=1,
       col=&quot;slateblue&quot;, ylim=c(0, ymx*1.02),
       main=i)
  abline(h=cutoff[1], col=&quot;red&quot;,lty=2) # 0.05 cutoff
  abline(h=cutoff[2], col=&quot;red&quot;,lty=1) # 0.1 cutoff
}
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>for(i in names(condi.m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx &lt;- maxlod(condi.m2.qtl.out[[i]]) # overall maximum LOD score
  plot(condi.m2.qtl.out[[i]], gm_DO2816_qc$gmap, lodcolumn=1,
       col=&quot;slateblue&quot;, ylim=c(0, ymx*1.02),
       main=i)
  abline(h=cutoff[1], col=&quot;red&quot;,lty=2) # 0.05 cutoff
  abline(h=cutoff[2], col=&quot;red&quot;,lty=1) # 0.1 cutoff
}</code></pre>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-1">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-1.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-1.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-1.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-2">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-2.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-2.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-2.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-3.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-3">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-3.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-3.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-3.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-4.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-4">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-4.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-4.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-4.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-5.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-5">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-5.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-5.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-5.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-6.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-6">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-6.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-6.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-6.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-7.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-7">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-7.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-7.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-7.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-8.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-8">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-8.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-8.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-8.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-9.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-9">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-9.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-9.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-9.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-10.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-10">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-10.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-10.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-10.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-11.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-11">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-11.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-11.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-12.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-12">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-12.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-12.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-12.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-13.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-13">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-13.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-13.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-13.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-14.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-14">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-14.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-14.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-15.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-15">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-15.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-15.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-15.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-16.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-16">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-16.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-16.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-16.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-17.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-17">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-17.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-17.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-17.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-18.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-18">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-18.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-18.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-18.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-19.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-19">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-19.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-19.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-20.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-20">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-20.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-20.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-21.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-21">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-21.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-21.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-22.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-22">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-22.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-22.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-22.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-23.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-23">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-23.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-23.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-23.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-24.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-24">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-24.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-24.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-24.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-25.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-25">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-25.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-25.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-26.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-26">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-26.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-26.png" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-26.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-27.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-27">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-27.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-27.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/13_plot_conditional_m2_qtlmapping.Rmd/plot%20scan%20result%20for%20conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-28.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-28">
Past versions of “plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-28.png”
</button>
</p>
<div id="fig-no-spaces-plot-scan-result-for-conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/figure/13_plot_conditional_m2_qtlmapping.Rmd/plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019-28.png" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#load blup results
x &lt;- list()
#colnames(pheno)[14:41]
for(i in colnames(pheno)[14:41]){
  y &lt;- get(load(paste0(&quot;output/blup/conditional.m2.blup_Novelty_residuals_RankNormal_datarelease_12182019_&quot;,i,&quot;.RData&quot;)))
  x[[i]] &lt;- y[[i]]
}
condi.m2.qtl.blup = x

#coeffects plot
condi.m2.qtl.peak &lt;- list()
pdf(file = paste0(&quot;output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019&quot;, &quot;_coeffgeneplot.pdf&quot;), width = 14)
#names(condi.m2.qtl.out)
for(i in names(condi.m2.qtl.out)){
  #peak
  condi.m2.qtl.peak[[i]] &lt;- find_peaks(condi.m2.qtl.out[[i]], gm_DO2816_qc$gmap,
                                       threshold=maxlod(condi.m2.qtl.out[[i]])-0.001,
                                       drop = 1.5)
  condi.m2.qtl.peak[[i]]$lodcolumn &lt;- i
  print(condi.m2.qtl.peak[[i]])
  
  chr &lt;- as.character(condi.m2.qtl.peak[[i]][condi.m2.qtl.peak[[i]]$lodcolumn == i,&quot;chr&quot;])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(condi.m2.coef[[i]],
               gm_DO2816_qc$gmap[chr],
               scan1_output=condi.m2.qtl.out[[i]],
               bgcolor=&quot;gray95&quot;, #legend=&quot;bottomleft&quot;,
               main =i)
  plot_coefCC(condi.m2.qtl.blup[[i]],
              gm_DO2816_qc$gmap[chr],
              scan1_output=condi.m2.qtl.out[[i]],
              bgcolor=&quot;gray95&quot;, #legend=&quot;bottomleft&quot;,
              main =i)
#   plot(condi.m2.snps[[i]]$lod,
#         condi.m2.snps[[i]]$snpinfo,
#         drop_hilit=1.5,
#         genes=condi.m2.genes[[i]],
#         main = i)
}</code></pre>
<pre><code>  lodindex                   lodcolumn chr    pos      lod ci_lo  ci_hi
1        1 OFA.total.distance.traveled  10 76.513 8.446601 73.92 77.149</code></pre>
<pre><code>  lodindex                   lodcolumn chr    pos     lod  ci_lo  ci_hi
1        1 OFA.distance.traveled.slope  18 16.723 7.81814 16.071 23.093</code></pre>
<pre><code>  lodindex                lodcolumn chr   pos      lod  ci_lo  ci_hi
1        1 OFA.total.time.in.corner   2 49.76 8.521552 49.072 50.588</code></pre>
<pre><code>  lodindex                   lodcolumn chr    pos      lod ci_lo  ci_hi
1        1 OFA.total.time.in.perimeter  10 74.106 9.784588 73.92 74.503</code></pre>
<pre><code>  lodindex                lodcolumn chr    pos      lod  ci_lo  ci_hi
1        1 OFA.total.time.in.center   6 46.518 8.016775 46.052 46.975</code></pre>
<pre><code>  lodindex                 lodcolumn chr    pos      lod ci_lo  ci_hi
1        1 OFA.total.ambulatory.time  10 73.985 8.645845 73.92 76.531</code></pre>
<pre><code>  lodindex                                lodcolumn chr    pos     lod
1        1 OFA.total.distance.traveled.in.perimeter  10 74.074 10.7076
  ci_lo  ci_hi
1 73.92 77.088</code></pre>
<pre><code>  lodindex                             lodcolumn chr    pos      lod ci_lo
1        1 OFA.total.distance.traveled.in.center   6 46.616 7.217382 2.535
   ci_hi
1 47.044</code></pre>
<pre><code>  lodindex                             lodcolumn chr    pos      lod
1        1 OFA.total.distance.traveled.in.corner  16 16.709 8.297126
   ci_lo  ci_hi
1 16.266 16.814</code></pre>
<pre><code>  lodindex           lodcolumn chr   pos      lod ci_lo  ci_hi
1        1 OFA.pct.dist.center   6 5.541 7.769937 3.226 51.574</code></pre>
<pre><code>  lodindex           lodcolumn chr    pos      lod  ci_lo  ci_hi
1        1 OFA.pct.dist.corner   6 51.573 7.688641 45.925 52.554</code></pre>
<pre><code>  lodindex              lodcolumn chr    pos      lod  ci_lo  ci_hi
1        1 OFA.pct.dist.perimeter   1 32.928 9.274424 31.684 33.569</code></pre>
<pre><code>  lodindex           lodcolumn chr    pos      lod  ci_lo  ci_hi
1        1 OFA.pct.time.center   1 33.365 8.028654 31.193 33.554</code></pre>
<pre><code>  lodindex           lodcolumn chr   pos      lod ci_lo  ci_hi
1        1 OFA.pct.time.corner   6 8.797 6.995772 5.276 46.975</code></pre>
<pre><code>  lodindex              lodcolumn chr    pos      lod  ci_lo  ci_hi
1        1 OFA.pct.time.perimeter   1 32.929 8.290732 31.582 33.554</code></pre>
<pre><code>  lodindex                        lodcolumn chr    pos      lod ci_lo
1        1 OFA.distance.traveled.first.five  10 73.985 11.43961 73.92
   ci_hi
1 74.248</code></pre>
<pre><code>  lodindex                       lodcolumn chr    pos      lod ci_lo ci_hi
1        1 OFA.distance.traveled.last.five   5 10.663 8.135841 10.51 11.96</code></pre>
<pre><code>  lodindex                         lodcolumn chr    pos      lod  ci_lo
1        1 LD.pct.distance.traveled.in.light  15 34.548 7.257033 33.831
   ci_hi
1 34.552</code></pre>
<pre><code>  lodindex                       lodcolumn chr    pos      lod ci_lo
1        1 LD.pct.ambulatory.time.in.light   8 23.078 6.753169 19.37
   ci_hi
1 24.497</code></pre>
<pre><code>  lodindex                    lodcolumn chr   pos      lod ci_lo ci_hi
1        1 LD.pct.resting.time.in.light  10 6.973 6.706699 5.865 7.626</code></pre>
<pre><code>  lodindex            lodcolumn chr    pos      lod  ci_lo  ci_hi
1        1 LD.pct.time.in.light  15 34.078 6.881954 32.361 34.552</code></pre>
<pre><code>  lodindex                         lodcolumn chr    pos      lod ci_lo
1        1 LD.pct.ambulatory.Counts.in.light   8 23.078 7.111158 19.37
   ci_hi
1 23.517</code></pre>
<pre><code>  lodindex            lodcolumn chr   pos      lod ci_lo ci_hi
1        1 LD.total.transitions   6 2.158 7.096609 1.923 2.253</code></pre>
<pre><code>  lodindex        lodcolumn chr    pos      lod ci_lo  ci_hi
1        1 HB.Total.Entries   2 48.993 7.758587 19.57 49.042</code></pre>
<pre><code>  lodindex        lodcolumn chr   pos      lod  ci_lo  ci_hi
1        1 HB.Novel.Entries  18 21.29 6.400137 21.017 21.893</code></pre>
<pre><code>  lodindex         lodcolumn chr    pos      lod  ci_lo  ci_hi
1        1 HB.Repeat.Entries   2 48.993 8.743605 48.974 49.042</code></pre>
<pre><code>  lodindex                                         lodcolumn chr    pos
1        1 NPP.NoveltyPreference.ZoneTime_WhiteVsBlack_Total   9 14.308
       lod  ci_lo  ci_hi
1 6.497207 11.625 14.321</code></pre>
<pre><code>  lodindex                                           lodcolumn chr   pos
1        1 NPP.NoveltyPreference.ZoneTime_GreyWhiteBlack_Total   5 3.515
       lod ci_lo ci_hi
1 6.573749 3.164 3.833</code></pre>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS release 6.5 (Final)

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.1.0 qtl2_0.18    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.2        pillar_1.3.1      git2r_0.23.0     
 [4] plyr_1.8.4        workflowr_1.4.0   tools_3.3.2      
 [7] digest_0.6.18     bit_1.1-14        RSQLite_2.1.1    
[10] evaluate_0.10     memoise_1.1.0     tibble_2.1.3     
[13] gtable_0.2.0      pkgconfig_2.0.1   rlang_0.4.0      
[16] DBI_1.0.0         yaml_2.2.0        parallel_3.3.2   
[19] withr_2.1.2       stringr_1.3.1     dplyr_0.8.3      
[22] knitr_1.20        fs_1.2.6          rprojroot_1.3-2  
[25] bit64_0.9-7       grid_3.3.2        tidyselect_0.2.5 
[28] glue_1.3.1        data.table_1.11.4 R6_2.4.0         
[31] rmarkdown_1.11    purrr_0.3.2       blob_1.1.1       
[34] magrittr_1.5      whisker_0.3-2     backports_1.1.2  
[37] scales_1.0.0      htmltools_0.3.6   assertthat_0.2.1 
[40] colorspace_1.4-0  stringi_1.2.4     lazyeval_0.2.1   
[43] munsell_0.5.0     crayon_1.3.4     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
