<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>License</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CSNA_workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/TheJacksonLaboratory/CSNA">GitHub</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/CSNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">License</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-11-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>csna_workflow/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryCSNAtree1b61cb68ff3b04985bb9ef687e6fd2df97acd234targetblank1b61cb6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/TheJacksonLaboratory/CSNA/tree/1b61cb68ff3b04985bb9ef687e6fd2df97acd234" target="_blank">1b61cb6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryCSNAtree1b61cb68ff3b04985bb9ef687e6fd2df97acd234targetblank1b61cb6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/TheJacksonLaboratory/CSNA/tree/1b61cb68ff3b04985bb9ef687e6fd2df97acd234" target="_blank">1b61cb6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/.nfs0000000135a08b9b00002448
    Untracked:  analysis/.nfs0000000135a08ba50000244a
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a1.R
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a1.err
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a1.out
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a1.sh
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a2.R
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a2.err
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a2.out
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a2.sh
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a3.R
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a3.err
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a3.out
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a3.sh
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a4.R
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a4.err
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a4.out
    Untracked:  analysis/19_qtl_analysis_RTG20_GS_a4.sh
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a1.R
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a1.err
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a1.out
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a1.sh
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a2.R
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a2.err
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a2.out
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a2.sh
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a3.R
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a3.err
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a3.out
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a3.sh
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a4.R
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a4.err
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a4.out
    Untracked:  analysis/19_qtl_analysis_RTG20_G_a4.sh
    Untracked:  analysis/csna_image.sif
    Untracked:  analysis/scripts/
    Untracked:  analysis/temp/
    Untracked:  analysis/workflow_proc.R
    Untracked:  analysis/workflow_proc.sh
    Untracked:  analysis/workflow_proc.stderr
    Untracked:  analysis/workflow_proc.stdout
    Untracked:  code/reconst_utils.R
    Untracked:  data/69k_grid_pgmap.RData
    Untracked:  data/FinalReport/
    Untracked:  data/GCTA/
    Untracked:  data/GM/
    Untracked:  data/Jackson_Lab_11_batches/
    Untracked:  data/Jackson_Lab_12_batches/
    Untracked:  data/Jackson_Lab_Bubier_MURGIGV01/
    Untracked:  data/Jackson_Lab_Gagnon/
    Untracked:  data/RTG/
    Untracked:  data/cc_variants.sqlite
    Untracked:  data/marker_grid_0.02cM_plus.txt
    Untracked:  data/mouse_genes_mgi.sqlite
    Untracked:  data/pheno/
    Untracked:  output/AfterQC_Percent_missing_genotype_data.pdf
    Untracked:  output/AfterQC_Proportion_matching_genotypes_after_removal_of_bad_samples.pdf
    Untracked:  output/AfterQC_Proportion_matching_genotypes_before_removal_of_bad_samples.pdf
    Untracked:  output/AfterQC_number_crossover.pdf
    Untracked:  output/DO_Gigamuga_chr2_WSB_G21.png
    Untracked:  output/DO_Gigamuga_chr2_WSB_G22.png
    Untracked:  output/DO_Gigamuga_chr2_WSB_G23.png
    Untracked:  output/DO_Gigamuga_chr2_WSB_G25.png
    Untracked:  output/DO_Gigamuga_chr2_WSB_G29.png
    Untracked:  output/DO_Gigamuga_chr2_WSB_G30.png
    Untracked:  output/DO_Gigamuga_chr2_WSB_G31.png
    Untracked:  output/DO_Gigamuga_chr2_WSB_G32.png
    Untracked:  output/DO_Gigamuga_chr2_WSB_G33.png
    Untracked:  output/DO_Gigamuga_founder_proportions_G21.png
    Untracked:  output/DO_Gigamuga_founder_proportions_G22.png
    Untracked:  output/DO_Gigamuga_founder_proportions_G23.png
    Untracked:  output/DO_Gigamuga_founder_proportions_G25.png
    Untracked:  output/DO_Gigamuga_founder_proportions_G29.png
    Untracked:  output/DO_Gigamuga_founder_proportions_G30.png
    Untracked:  output/DO_Gigamuga_founder_proportions_G31.png
    Untracked:  output/DO_Gigamuga_founder_proportions_G32.png
    Untracked:  output/DO_Gigamuga_founder_proportions_G33.png
    Untracked:  output/DO_Gigamuga_founder_proportions_chr2_G21.png
    Untracked:  output/DO_Gigamuga_founder_proportions_chr2_G22.png
    Untracked:  output/DO_Gigamuga_founder_proportions_chr2_G23.png
    Untracked:  output/DO_Gigamuga_founder_proportions_chr2_G25.png
    Untracked:  output/DO_Gigamuga_founder_proportions_chr2_G29.png
    Untracked:  output/DO_Gigamuga_founder_proportions_chr2_G30.png
    Untracked:  output/DO_Gigamuga_founder_proportions_chr2_G31.png
    Untracked:  output/DO_Gigamuga_founder_proportions_chr2_G32.png
    Untracked:  output/DO_Gigamuga_founder_proportions_chr2_G33.png
    Untracked:  output/DO_recom_block_size_G21.png
    Untracked:  output/DO_recom_block_size_G22.png
    Untracked:  output/DO_recom_block_size_G23.png
    Untracked:  output/DO_recom_block_size_G25.png
    Untracked:  output/DO_recom_block_size_G29.png
    Untracked:  output/DO_recom_block_size_G30.png
    Untracked:  output/DO_recom_block_size_G31.png
    Untracked:  output/DO_recom_block_size_G32.png
    Untracked:  output/DO_recom_block_size_G33.png
    Untracked:  output/Histgram_Prj01_RL-Acquisition_preqc_02142020.pdf
    Untracked:  output/Histgram_Prj01_RL-Reversal_preqc_02142020.pdf
    Untracked:  output/Histgram_Prj02_Sensitization_preqc_02142020.pdf
    Untracked:  output/Histgram_Prj04_IVSA_preqc_02142020.pdf
    Untracked:  output/Novelty_residuals_RankNormal_datarelease_12182019-69k-genelist.csv
    Untracked:  output/Novelty_residuals_RankNormal_datarelease_12182019-69k-variantlist.csv
    Untracked:  output/Novelty_residuals_RankNormal_datarelease_12182019-genelist.csv
    Untracked:  output/Novelty_residuals_RankNormal_datarelease_12182019-variantlist.csv
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted.pdf
    Untracked:  output/Percent_missing_genotype_data.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker.pdf
    Untracked:  output/Proportion_matching_genotypes_after_removal_of_bad_samples.pdf
    Untracked:  output/Proportion_matching_genotypes_after_removal_samples_percent_missing_5.pdf
    Untracked:  output/Proportion_matching_genotypes_before_removal_of_bad_samples.pdf
    Untracked:  output/Proportion_matching_genotypes_before_removal_samples.pdf
    Untracked:  output/RTG/
    Untracked:  output/blup/
    Untracked:  output/condi.m2.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/condi.m2.qtl.blup_Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/condi.m2.qtl.out.RData
    Untracked:  output/conditional.69k.Novelty_residuals_RankNormal_datarelease_12182019-genelist.csv
    Untracked:  output/conditional.69k.Novelty_residuals_RankNormal_datarelease_12182019-variantlist.csv
    Untracked:  output/conditional.Novelty_residuals_RankNormal_datarelease_12182019-genelist.csv
    Untracked:  output/conditional.Novelty_residuals_RankNormal_datarelease_12182019-variantlist.csv
    Untracked:  output/conditional.m2.69k.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/conditional.m2.69k.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/conditional.m2.69k.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/conditional.m2.qtl.out.RData
    Untracked:  output/conditional_m2.Novelty_residuals_RankNormal_datarelease_07302918_genomescan.pdf
    Untracked:  output/conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/genotype_error_marker.pdf
    Untracked:  output/genotype_frequency_marker.pdf
    Untracked:  output/m1.Novelty_resids_datarelease_07302918.RData
    Untracked:  output/m1.Novelty_resids_datarelease_07302918_coeffgeneplot.pdf
    Untracked:  output/m1.Novelty_resids_datarelease_07302918_genomescan.pdf
    Untracked:  output/m1.Novelty_resids_datarelease_12182019.RData
    Untracked:  output/m1.Novelty_resids_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m1.Novelty_resids_datarelease_12182019_genomescan.pdf
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_07302918.RData
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_07302918_genomescan.pdf
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m1.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/m2.69k.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/m2.69k.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m2.69k.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/m2.69k.Prj01_RL-Acquisition_preqc_02142020.RData
    Untracked:  output/m2.69k.Prj01_RL-Reversal_preqc_02142020.RData
    Untracked:  output/m2.69k.Prj02_Sensitization_preqc_02142020.RData
    Untracked:  output/m2.69k.Prj04_IVSA_preqc_02142020.RData
    Untracked:  output/m2.Novelty_resids_datarelease_07302918.RData
    Untracked:  output/m2.Novelty_resids_datarelease_07302918_genomescan.pdf
    Untracked:  output/m2.Novelty_resids_datarelease_12182019.RData
    Untracked:  output/m2.Novelty_resids_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m2.Novelty_resids_datarelease_12182019_genomescan.pdf
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_07302918.RData
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_07302918_genomescan.pdf
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m2.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/m2.Prj01_RL-Acquisition_preqc_02142020.RData
    Untracked:  output/m2.Prj01_RL-Acquisition_preqc_02142020_qtl2scan.pdf
    Untracked:  output/m2.Prj01_RL-Reversal_preqc_02142020.RData
    Untracked:  output/m2.Prj01_RL-Reversal_preqc_02142020_qtl2scan.pdf
    Untracked:  output/m2.Prj02_Sensitization_preqc_02142020.RData
    Untracked:  output/m2.Prj02_Sensitization_preqc_02142020_qtl2scan.pdf
    Untracked:  output/m2.Prj04_IVSA_preqc_02142020.RData
    Untracked:  output/m2.Prj04_IVSA_preqc_02142020_qtl2scan.pdf
    Untracked:  output/m3.Novelty_resids_datarelease_07302918.RData
    Untracked:  output/m3.Novelty_resids_datarelease_07302918_genomescan.pdf
    Untracked:  output/m3.Novelty_resids_datarelease_12182019.RData
    Untracked:  output/m3.Novelty_resids_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m3.Novelty_resids_datarelease_12182019_genomescan.pdf
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_07302918.RData
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_07302918_genomescan.pdf
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_12182019.RData
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_12182019_coeffgeneplot.pdf
    Untracked:  output/m3.Novelty_residuals_RankNormal_datarelease_12182019_genomescan.pdf
    Untracked:  output/num.geno.pheno.in.Novelty_resids_datarelease_07302918.csv
    Untracked:  output/num.geno.pheno.in.Novelty_resids_datarelease_12182019.csv
    Untracked:  output/num.geno.pheno.in.Novelty_residuals_RankNormal_datarelease_07302918.csv
    Untracked:  output/num.geno.pheno.in.Novelty_residuals_RankNormal_datarelease_12182019.csv
    Untracked:  output/number_crossover.pdf
    Untracked:  output/permu/
    Untracked:  output/prop_across_generation_chr_p.RData

Unstaged changes:
    Modified:   README.md
    Modified:   _workflowr.yml
    Deleted:    analysis/01_geneseek2qtl2.R
    Deleted:    analysis/02_geneseek2intensity.R
    Deleted:    analysis/03_firstgm2genoprobs.R
    Deleted:    analysis/04_diagnosis_qc_gigamuga_11_batches.Rmd
    Deleted:    analysis/04_diagnosis_qc_gigamuga_nine_batches.R
    Deleted:    analysis/04_diagnosis_qc_gigamuga_nine_batches.Rmd
    Deleted:    analysis/05_after_diagnosis_qc_gigamuga_11_batches.Rmd
    Deleted:    analysis/05_after_diagnosis_qc_gigamuga_nine_batches.R
    Deleted:    analysis/05_after_diagnosis_qc_gigamuga_nine_batches.Rmd
    Deleted:    analysis/06_final_pr_apr_69K.R
    Deleted:    analysis/07.1_html_founder_prop.R
    Deleted:    analysis/07_recomb_size_founder_prop.R
    Deleted:    analysis/07_recomb_size_founder_prop.Rmd
    Deleted:    analysis/08_gcta_herit.R
    Deleted:    analysis/09_qtlmapping.R
    Deleted:    analysis/10_qtl_permu.R
    Deleted:    analysis/11_qtl_blup.R
    Deleted:    analysis/12_plot_69k_qtl_mapping_2.Rmd
    Deleted:    analysis/12_plot_qtl_mapping_1.Rmd
    Deleted:    analysis/12_plot_qtl_mapping_2.Rmd
    Deleted:    analysis/13_plot_69k_conditional_m2_qtlmapping.Rmd
    Deleted:    analysis/13_plot_conditional_m2_qtlmapping.Rmd
    Deleted:    analysis/16_diagnosis_qc_gigamuga_gagnon.Rmd
    Deleted:    analysis/16_diagnosis_qc_gigamuga_gagnon2.Rmd
    Deleted:    analysis/17_plot_qtl_mapping.Rmd
    Deleted:    analysis/run_01_geneseek2qtl2.R
    Deleted:    analysis/run_02_geneseek2intensity.R
    Deleted:    analysis/run_03_firstgm2genoprobs.R
    Deleted:    analysis/run_04_diagnosis_qc_gigamuga_nine_batches.R
    Deleted:    analysis/run_05_after_diagnosis_qc_gigamuga_nine_batches.R
    Deleted:    analysis/run_06_final_pr_apr_69K.R
    Deleted:    analysis/run_07.1_html_founder_prop.R
    Deleted:    analysis/run_07_recomb_size_founder_prop.R
    Deleted:    analysis/run_08_gcta_herit.R
    Deleted:    analysis/run_09_qtlmapping.R
    Deleted:    analysis/run_10_qtl_permu.R
    Deleted:    analysis/run_11_qtl_blup.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/license.Rmd</code>) and HTML (<code>docs/license.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/2ef67475f28da629414ef77d043e2c130d1e41a4/docs/license.html" target="_blank">2ef6747</a>
</td>
<td>
xhyuo
</td>
<td>
2020-11-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/c1326cdea35f9462c40409304246ea11c4b65f68/docs/license.html" target="_blank">c1326cd</a>
</td>
<td>
xhyuo
</td>
<td>
2020-11-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/4794cfedac6708a9ce38642c5bd469581c5ac6f3/docs/license.html" target="_blank">4794cfe</a>
</td>
<td>
xhyuo
</td>
<td>
2020-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/d8fe998c3d55e8a8da88739ba3ded3d9a0d8dd8a/docs/license.html" target="_blank">d8fe998</a>
</td>
<td>
xhyuo
</td>
<td>
2020-10-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/27b48ffb25e42cc59c12711b0de6c53de5e4abe3/docs/license.html" target="_blank">27b48ff</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/0404d4411cd581def8b06b7f0b753c015230f2a6/docs/license.html" target="_blank">0404d44</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/4aeac6f5a16fcb20e2fa1f4c05d577312943e297/docs/license.html" target="_blank">4aeac6f</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/a8606a225eb951cd7513de9927b3971c4e0ba656/docs/license.html" target="_blank">a8606a2</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/693e807016791180b33d8e668d14f49aec7c5269/docs/license.html" target="_blank">693e807</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/00d2604682fde29299907e949b3719bb1da9da09/docs/license.html" target="_blank">00d2604</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/0632c4a771882f2b5db5919cf1dbd1ce86c114df/docs/license.html" target="_blank">0632c4a</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/a0daa3a883574fe5c55de9b80c5ef641a480be1c/docs/license.html" target="_blank">a0daa3a</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/481cb4d2b381005587fcfbbcb349a866b51f1011/docs/license.html" target="_blank">481cb4d</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/c4f4559cf0c3fdb1320fd5430ca9d5b0acf27957/docs/license.html" target="_blank">c4f4559</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/d977cdc19eab6afa2b4d05fc9e88456424436b82/docs/license.html" target="_blank">d977cdc</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/6861ae73fa9c54567eec7190f86b62f9827f303f/docs/license.html" target="_blank">6861ae7</a>
</td>
<td>
xhyuo
</td>
<td>
2020-02-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/a2f202a3bf3f23ec2323e324d0b64fbbaccecae0/docs/license.html" target="_blank">a2f202a</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/1b2c84993f1fa4f603de0f53015eebedf167549c/docs/license.html" target="_blank">1b2c849</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/c5ca735760c442dba355e805532d83983a13e0c6/docs/license.html" target="_blank">c5ca735</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/626668d6004cd9658e35319050fa2fa8d28b608e/docs/license.html" target="_blank">626668d</a>
</td>
<td>
xhyuo
</td>
<td>
2020-01-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/8d8f9b4230e34e3065e3359fcf3fdce76f2f5508/docs/license.html" target="_blank">8d8f9b4</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/2c6b5635d202dd938105c4334a64cdcb63fff372/docs/license.html" target="_blank">2c6b563</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/81802a2660d2a340a8ca751226b35d873e8a9f58/docs/license.html" target="_blank">81802a2</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/a7547b3d45a62d6db3b0d16516020bd75220d4ff/docs/license.html" target="_blank">a7547b3</a>
</td>
<td>
xhyuo
</td>
<td>
2019-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/efed06b90051d37b076aa4728682439356b6d8c7/docs/license.html" target="_blank">efed06b</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/f187b9966748583771866baa95e9cd4340805056/docs/license.html" target="_blank">f187b99</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/d25bab6450bafb744e64c878fa0311710260a1f8/docs/license.html" target="_blank">d25bab6</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/58d4c4e21b62b1859e20fbd760e8d6eda584a733/docs/license.html" target="_blank">58d4c4e</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/51b872f7ff81e94c05406e4e0994753d3cfeb4a8/docs/license.html" target="_blank">51b872f</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/a6d688627c717e885ff4355857bed5f52edaf577/docs/license.html" target="_blank">a6d6886</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/e899cf15008af4993a62d5de2732bf9e32127c52/docs/license.html" target="_blank">e899cf1</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/7b693e9bd4fe46c62467c257970719255f77b359/docs/license.html" target="_blank">7b693e9</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/463afa3190e76a21b2a0beedd4a561d97e5fbfe2/docs/license.html" target="_blank">463afa3</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/a28dfe6d698866b9d5ebdb3f096ed40c1f1e8122/docs/license.html" target="_blank">a28dfe6</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/e79d270039d8e2458add5233f8e728ccfbcdf624/docs/license.html" target="_blank">e79d270</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/13d7e2957ee94a88b85dbf82462c017d6f463263/docs/license.html" target="_blank">13d7e29</a>
</td>
<td>
xhyuo
</td>
<td>
2019-10-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/c3ee11a150f3eedf5cf3c7fede84c3fbfaaacd3c/docs/license.html" target="_blank">c3ee11a</a>
</td>
<td>
xhyuo
</td>
<td>
2019-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/97188bc6892400711418899a25e52e3ef0ebed6e/docs/license.html" target="_blank">97188bc</a>
</td>
<td>
xhyuo
</td>
<td>
2019-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/0d002796967eadfd8e12dbb6ab0b27965feaa825/docs/license.html" target="_blank">0d00279</a>
</td>
<td>
xhyuo
</td>
<td>
2019-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/27c1d860b515e69ec25e7275872ccb757954ae89/docs/license.html" target="_blank">27c1d86</a>
</td>
<td>
xhyuo
</td>
<td>
2019-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/e27133058fbeb9671f5e5658bb14fbb488489e9c/docs/license.html" target="_blank">e271330</a>
</td>
<td>
xhyuo
</td>
<td>
2019-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/CSNA/1e4e3f711b8f33d111258c586c50a2382ab84142/docs/license.html" target="_blank">1e4e3f7</a>
</td>
<td>
xhyuo
</td>
<td>
2019-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/CSNA/blob/898ff80e7a88c046ed3b5a1fabac19eaf3a71ef3/analysis/license.Rmd" target="_blank">898ff80</a>
</td>
<td>
xhyuo
</td>
<td>
2019-09-22
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>What license are you using for your code? See <a href="http://choosealicense.com/">choosealicense.com</a> for help deciding. It’s a convention to save a file <code>LICENSE</code> in the root of your Git repo that contains the license text.</p>
<p>What license are you using for the written content on your site? It is traditional to choose a <a href="https://creativecommons.org/choose/">Creative Commons</a> license for this type of content.</p>
<p>How should others cite your work? It’s a convention to save a file <code>CITATION</code> in the root of your Git repo that contains the citation information.</p>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
