---
title: "DO_Diversity_Report"
author: "Hao He"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Diversity report for diversity outbred mice
After finishing 06_final_pr_apr_69K.R, 07_do_diversity_report.R, all the output will be used plot DO Diversity Report for 12 batches of DO mice

# library
```{r library, message=FALSE}
# Load packages
library(qtl2)
library(table1)
library(tidyverse)
library(data.table)
library(foreach)
library(doParallel)
library(parallel)
library(abind)
library(gap)
library(regress)
library(lme4)
library(abind)
library(ggplot2)
library(vcd)
library(MASS)
#library(plotly)
library(colorspace)
library(HardyWeinberg)
options(stringsAsFactors = FALSE)
source("code/reconst_utils.R")
```

#Summary
```{r Summary}
load("data/Jackson_Lab_12_batches/gm_DO3173_qc.RData")#gm_after_qc

# make dataset with a few variables to summarize
table1 <- gm_after_qc$covar %>% 
  dplyr::select(Name = name, 
                Sex  = sex, 
                Generation = ngen) %>%
  mutate(Sex = case_when(
    Sex == "F" ~ "Female",
    Sex == "M" ~ "Male"
  ))

# summarize the data 
table1(~ Generation | Sex, data=table1)
```

#Founder contributions
```{r Founder contributions, fig.height=12, fig.width=14, eval=TRUE}
load("data/Jackson_Lab_12_batches/fp_DO3173.RData") #fp and fp_summary object
#change order of level in gen
fp$gen <- factor(fp$gen,levels = c(21,22,23,25,29,30,31,32,33,34,35,36))

#summarize per generation per chromosome
fp_summary = fp %>% group_by(chr, founder, gen) %>%
  summarize(mean = round(100*mean(prop), 2),
            sd   = round(100*sd(prop), 2))
  
#Stackbar plot
#summarize per chromosome across generation
pdf(file = "data/Jackson_Lab_12_batches/stackbar_mean_prop_across_all_gen.pdf",width = 16)
p01 <- fp %>% group_by(chr, founder) %>%
  summarise(grand_mean = round(100*mean(prop), 2)) %>%
  ggplot(aes(x = chr, y = grand_mean, fill = founder)) +
    geom_bar(stat="identity",
             width=1) +
    geom_text(aes(label = paste0(grand_mean)), position = position_stack(vjust = 0.5)) +
    scale_fill_manual(values = CCcolors) +
    ylab("Mean percentage across generations") +
    xlab("Chromosome") +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          axis.line = element_line(colour = "black"),
          plot.title = element_text(hjust = 0.5),
          text = element_text(size=16),
          axis.title=element_text(size=16),
          legend.title=element_blank())
p01
dev.off()
p01

#Stackbar plot
#summarize per chromosome across generation
pdf(file = "data/Jackson_Lab_12_batches/stackbar_mean_prop_across_all_chr.pdf",width = 16)
p02 <- fp %>% group_by(gen, founder) %>%
  summarise(grand_mean = round(100*mean(prop), 2), 
            grand_sd   = round(100*sd(prop), 2)) %>%
  ggplot(aes(x = gen, y = grand_mean, fill = founder)) +
    geom_bar(stat="identity",
             width=0.99) +
    geom_text(aes(label = paste0(grand_mean, " ± ", grand_sd)), position = position_stack(vjust = 0.5)) +
    scale_fill_manual(values = CCcolors) +
    ylab("Mean percentage across all chromosomes") +
    xlab("Generation") +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          axis.line = element_line(colour = "black"),
          plot.title = element_text(hjust = 0.5),
          text = element_text(size=16),
          axis.title=element_text(size=16),
          legend.title=element_blank())
p02
dev.off()
p02

#stackbar_prop_across_gen
for(c in c(1:19, "X")){
  #print(c)
  p <- ggplot(data = fp_summary[fp_summary$chr == c,], aes(x = gen, y = mean, fill = founder)) +
    geom_bar(stat="identity",
             width=1) +
    geom_text(aes(label = paste0(mean," ± ", sd)), position = position_stack(vjust = 0.5)) +
    scale_fill_manual(values = CCcolors) +
    labs(title = paste0("Chr ", c)) +
    ylab("Percentage") +
    xlab("Generation") +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          axis.line = element_line(colour = "black"),
          plot.title = element_text(hjust = 0.5),
          text = element_text(size=16),
          axis.title=element_text(size=16),
          legend.title=element_blank())
  print(p)
}

pdf(file = "data/Jackson_Lab_12_batches/stackbar_prop_across_gen.pdf",width = 16)
for(c in c(1:19, "X")){
  #print(c)
  p <- ggplot(data = fp_summary[fp_summary$chr == c,], aes(x = gen, y = mean, fill = founder)) +
    geom_bar(stat="identity",
             width=1) +
    geom_text(aes(label = paste0(mean," ± ", sd)), position = position_stack(vjust = 0.5)) +
    scale_fill_manual(values = CCcolors) +
    labs(title = paste0("Chr ", c)) +
    ylab("Percentage") +
    xlab("Generation") +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          axis.line = element_line(colour = "black"),
          plot.title = element_text(hjust = 0.5),
          text = element_text(size=16),
          axis.title=element_text(size=16),
          legend.title=element_blank())
  print(p)
}
dev.off()

#stackbar_prop_across_chr
for(g in levels(fp_summary$gen)){
  #print(g)
  p <- ggplot(data = fp_summary[fp_summary$gen == g,], aes(x = chr, y = mean, fill = founder)) +
    geom_bar(stat="identity",
             width=1) +
    geom_text(aes(label = paste0(mean)), position = position_stack(vjust = 0.5)) +
    scale_fill_manual(values = CCcolors) +
    labs(title = paste0("Generation ", g)) +
    ylab("Percentage") +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          axis.title.x = element_blank(),
          axis.line = element_line(colour = "black"),
          plot.title = element_text(hjust = 0.5),
          text = element_text(size=16),
          axis.title=element_text(size=16),
          legend.title=element_blank())
  #print(p)
}

pdf(file = "data/Jackson_Lab_12_batches/stackbar_prop_across_chr.pdf", width = 12)
for(g in levels(fp_summary$gen)){
  #print(g)
  p <- ggplot(data = fp_summary[fp_summary$gen == g,], aes(x = chr, y = mean, fill = founder)) +
    geom_bar(stat="identity",
             width=1) +
    geom_text(aes(label = paste0(mean)), position = position_stack(vjust = 0.5)) +
    scale_fill_manual(values = CCcolors) +
    labs(title = paste0("Generation ", g)) +
    ylab("Percentage") +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          axis.title.x = element_blank(),
          axis.line = element_line(colour = "black"),
          plot.title = element_text(hjust = 0.5),
          text = element_text(size=16),
          axis.title=element_text(size=16),
          legend.title=element_blank())
  print(p)
}
dev.off()

#line plot
#plt <- htmltools::tagList()
for(c in unique(names(gm_after_qc$geno))){
  print(c)
  fp_subdata <- fp[fp$chr == c,]
  pp <- ggplot(data = fp_subdata,aes(pos, prop, group = gen, color = founder)) +
    geom_line(aes(linetype=gen)) +
    scale_linetype_manual(values=rep("solid",12)) +
    geom_hline(yintercept=0.125, linetype="dashed", color = "black", size = 0.25) +
    scale_color_manual(values = CCcolors) +
    facet_grid(founder~.) + 
    labs(title = paste0("Chr ", c)) + 
    theme(legend.position='none')
  print(pp)
  # Print an interactive plot
  # Add to list
  #plt[[c]] <- as_widget(ggplotly(pp, width = 1000, height = 1000))
}
#plt
```

#Average haplotype block size
```{r Average haplotype block size, fig.width=10, fig.height=8, eval=T}
load("data/Jackson_Lab_12_batches/recom_block_size.RData")
#Create an appropriately sized vector of names
nameVector <- unlist(mapply(function(x,y){ rep(y, length(x)) }, pos_ind_gen, names(pos_ind_gen)))
#Create the result
recom_block <- cbind.data.frame(unlist(pos_ind_gen), nameVector)
colnames(recom_block) <- c("sizeblock",
                           "ngen")
#remove 0
recom_block <- recom_block[recom_block$sizeblock != 0,]
recom_block$ngen <- factor(recom_block$ngen, levels = as.character(c(21:36)))
#mean
means <- aggregate(sizeblock~ngen, data= recom_block,mean)
means$sizeblock <- round(means$sizeblock, 2)

pdf(file = "data/Jackson_Lab_12_batches/boxplot_mean_recomb_block_size.pdf", height = 8, width = 10)
p1 <- ggplot(recom_block, aes(x=ngen, y=sizeblock, group = ngen, fill = ngen)) + 
  geom_boxplot(show.legend = F , outlier.size = 0.5, notchwidth = 3) +
  scale_x_discrete(drop=FALSE, breaks = c(21:23,NA,25,rep(NA,3),29:36)) +
  scale_fill_discrete_qualitative(palette = "warm")+
  geom_text(data = means, alpha = 0.85, aes(label = sizeblock, y = sizeblock + 0.15 )) + 
  ylab("Recombination Block Size (Mb)") +
  xlab("Generation") +
  labs(fill = "") +
  #ylim(c(0, 60)) +
  scale_y_continuous(breaks=c(0,5,10, 20, 40, 60), limits=c(0, 60)) +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(), 
        axis.line = element_line(colour = "black"),
        text = element_text(size=16), 
        axis.title=element_text(size=16)) +
  guides(shape = guide_legend(override.aes = list(size = 12)))
p1
dev.off()
p1

#block size distribution
for(g in unique(gm_after_qc$covar$ngen)){
  #plot for recom block size
  #png(paste0("data/Jackson_Lab_12_batches/DO_recom_block_size_G", g, ".png"))
  x <- pos_ind_gen[[g]][pos_ind_gen[[g]] != 0]
  # estimate the parameters
  fit1 <- fitdistr(x, "exponential") 
  
  # goodness of fit test
  ks.test(x, "pexp", fit1$estimate) # p-value > 0.05 -> distribution not refused
  
  # plot a graph
  hist(x, 
       freq = FALSE, 
       breaks = 200, 
       xlim = c(0, 5+quantile(x, 1)), 
       #ylim = c(0,0.3),
       xlab = "Recombination Block Size (Mb)", 
       main = paste0("Gen ", g))
  curve(dexp(x, rate = fit1$estimate), 
        from = 0, 
        to = 5+quantile(x, 1), 
        col = "red", 
        add = TRUE)
  #dev.off()
}
```

#Average heterozygosity value
```{r Average heterozygosity value, fig.width=10, fig.height=8, eval=T}
load("data/Jackson_Lab_12_batches/dat_het_ind_pr.RData")
dat_het_ind_pr$ngen <- factor(dat_het_ind_pr$ngen, levels = as.character(c(21:36)))

pdf(paste("data/Jackson_Lab_12_batches/DO_Heterozygosity_value_violin_genoprops.pdf"), width = 10, height =8)
p2 <- ggplot(dat_het_ind_pr, aes(x=ngen, y=het, group=ngen, fill=ngen)) +
  geom_violin(show.legend = FALSE) +
  geom_boxplot(show.legend = FALSE, width=0.35, color="black", alpha=0.6) +
  scale_x_discrete(drop=FALSE, breaks = c(21:23,NA,25,rep(NA,3),29:36)) +
  scale_fill_discrete_qualitative(palette = "warm")+
  ylab("Heterozygosity from genotype props") +
  xlab("Generation") +
  ylim(c(0.65, 1)) +
  geom_hline(yintercept=0.875, linetype="dashed", color = "red") +
  #scale_y_continuous(breaks=c(0.55, 0.65, 0.75, 0.85, 0.95, 1), limits=c(0.55, 1)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black"),
        text = element_text(size=16),
        axis.title=element_text(size=16),
        legend.title=element_blank())
p2
dev.off()
p2
```

#marker UNC13316610 allele frequency
```{r marker UNC13316610 allele frequency, fig.width=10, fig.height=8, eval=T, warning = FALSE,message=FALSE}
T_freq <- unique(gm_after_qc$covar$ngen) %>%
  map(~(read.table(paste0("data/GCTA/12_batches_QC_id_gen", .x,".frq"), header = TRUE) %>%
          filter(SNP == "UNC13316610") %>%
          mutate(GEN = .x, .before = 1))
        ) %>%
  set_names(., nm = unique(gm_after_qc$covar$ngen)) %>%
  bind_rows() %>%
  mutate(allele = "T",
         MAF_allele = case_when(
           A1 == "T" ~ MAF,
           A1 != "T" ~ 1-MAF
         ))
C_freq <- T_freq %>%
  mutate(allele = "C",
         MAF_allele = 1-MAF_allele
         )
#bind
UNC13316610.freq <- bind_rows(T_freq, C_freq)
# Stacked
p <- ggplot(UNC13316610.freq, aes(fill=allele, y=MAF_allele, x=GEN)) + 
  geom_bar(position="stack", stat="identity") +
  scale_fill_manual(values=c("#3399CC", "#FFCC33")) +
  xlab("Generation") +
  ylab("UNC13316610 allele frequency")
print(p)

#pull marker UNC13316610
gm.UNC13316610 <- pull_markers(gm_after_qc, "UNC13316610")
#genotype freq
gfreq <- calc_raw_geno_freq(gm.UNC13316610)
gfreq_tab <- gfreq %>%
  as.data.frame() %>%
  mutate(id = rownames(gfreq)) %>%
  left_join(gm_after_qc$covar) %>%
  group_by(ngen)

#gfreq_tab
tab <- gfreq_tab %>%
  group_map(~(apply(.x[,1:3],2,function(x)(sum(x,na.rm = T))))) %>%
  bind_rows() %>%
  mutate(Genertation = group_keys(gfreq_tab)$ngen, .before = 1) %>%
  left_join(C_freq[,c("GEN", "MAF_allele")], by = c("Genertation" = "GEN")) %>%
  left_join(T_freq[,c("GEN", "MAF_allele")], by = c("Genertation" = "GEN")) %>%
  rename(C_freq = MAF_allele.x) %>%
  rename(T_freq = MAF_allele.y) %>%
  mutate(C_freq2 = (2*AA + AB)/(2*(AA+AB+BB))) %>%
  rowwise() %>%
  mutate(chisq = HWChisq(c(AA, AB, BB))$chisq,
         df    = HWChisq(c(AA, AB, BB))$df,
         p     = HWChisq(c(AA, AB, BB))$p) %>%
  dplyr::select(Genertation = Genertation,
                CC = AA,
                CT = AB,
                TT = BB,
                C_freq,
                T_freq,
                chisq,
                p)

tab
```



