---
title: "plot_qtl_69k_mapping_Novelty_residuals_RankNormal_datarelease_12182019"
author: "Hao He"
date: "`r Sys.Date()`"
output: html_document
---
  
  ```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This script will plot the qtl mapping results.

# library
```{r library, message=FALSE}
library(qtl2)
library(ggplot2)
load("data/Jackson_Lab_11_batches/gm_DO2816_qc.RData")
load("data/69k_grid_pgmap.RData")
pheno <- read.csv("data/pheno/Novelty_residuals_RankNormal_datarelease_12182019.csv", header = TRUE)
```
# plot 69k scan result for m2 Novelty_residuals_RankNormal_datarelease_12182019
```{r plot 69k scan result for m2 Novelty_residuals_RankNormal_datarelease_12182019,fig.height=14, fig.width=14, eval=TRUE}
print("m2")
load("output/m2.69k.Novelty_residuals_RankNormal_datarelease_12182019.RData")

#permutation results
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.
operm <- list()
for(i in c(1:50)[c(-4,-8,-14,-15,-18,-28,-31)]){
  operm[[i]] <- get(load(paste0("output/permu/m2_69k_Novelty_residuals_RankNormal_datarelease_12182019_", colnames(pheno)[14], "_", i, ".RData")))
  operm[[i]] <- operm[[i]]$OFA.total.distance.traveled
}
operm <- matrix(do.call(rbind, operm[c(1:50)[c(-4,-8,-14,-15,-18,-28,-31)]]),ncol = 1)
class(operm) <- c("scan1perm", "matrix")
cutoff <- summary(operm, c(0.05, 0.1))[,1]

#genome-wide results
pdf(file = paste0("output/m2.69k.Novelty_residuals_RankNormal_datarelease_12182019", "_genomescan.pdf"), width = 14)
for(i in names(m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m2.qtl.out[[i]]) # overall maximum LOD score
  plot(m2.qtl.out[[i]], gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
  abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
}
dev.off()

#for(i in names(m2.qtl.out)){
#  par(mar=c(5.1, 4.1, 1.1, 1.1))
#  ymx <- maxlod(m2.qtl.out[[i]]) # overall maximum LOD score
#  plot(m2.qtl.out[[i]], gmap, lodcolumn=1,
#       col="slateblue", ylim=c(0, ymx*1.02),
#       main=i)
#  abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
#  abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
#}

#load blup results
x <- list()
for(i in colnames(pheno)[14:41]){
  y <- get(load(paste0("output/blup/m2_blup_69k_Novelty_residuals_RankNormal_datarelease_12182019_",i,".RData")))
  x[[i]] <- y[[i]]
}
m2.blup = x

#coeffects plot
pdf(file = paste0("output/m2.69k.Novelty_residuals_RankNormal_datarelease_12182019", "_coeffgeneplot.pdf"), width = 14)
for(i in names(m2.qtl.out)){
  chr <- as.character(m2.peak[[i]][m2.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  #  plot_coefCC(m2.coef[[i]],
  #              gmap[chr],
  #              scan1_output=m2.qtl.out[[i]],
  #              bgcolor="gray95", #legend="bottomleft",
  #              main =i)
  plot_coefCC(m2.blup[[i]],
              gmap[chr],
              scan1_output=m2.qtl.out[[i]],
              bgcolor="gray95", #legend="bottomleft",
              main =i)
#   plot(m2.snps[[i]]$lod,
#        m2.snps[[i]]$snpinfo,
#        drop_hilit=1.5,
#        genes=m2.genes[[i]],
#        main = i)
}
dev.off()

for(i in names(m2.qtl.out)){
  print(i)
  if(is.null(m2.snps[[i]]))
    next

  if(dim(m2.genes[[i]]) == 0){
    m2.genes[[i]] = NA
  }

  top_1 <- data.frame(pheno = i)
  top <- top_snps(m2.snps[[i]]$lod, m2.snps[[i]]$snpinfo)
  top <- cbind(top_1, top)
  write.table(top, file = "output/Novelty_residuals_RankNormal_datarelease_12182019-69k-variantlist.csv", sep = ",",
              row.names = F, col.names = T, append = TRUE)
  
  topgene <- cbind(top_1, m2.genes[[i]])
  write.table(topgene, file = "output/Novelty_residuals_RankNormal_datarelease_12182019-69k-genelist.csv", sep = ",",
              row.names = F, col.names = T, append = TRUE)
}
```
