---
title: "plot_conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019"
author: "Hao He"
date: "`r Sys.Date()`"
output: html_document
---

# plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019
```{r plot scan result for conditional_m2.Novelty_residuals_RankNormal_datarelease_12182019, fig.height=14, fig.width=14, eval=TRUE}
#   This script performs plotting on qtl conditional mapping for DO mice m2, Novelty_residuals_RankNormal_datarelease_12182019
library(qtl2)
library(ggplot2)
load("data/Jackson_Lab_11_batches/gm_DO2816_qc.RData")
pheno <- read.csv("data/pheno/Novelty_residuals_RankNormal_datarelease_12182019.csv", header = TRUE)

print("m2")
load("output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019.RData")

#permutation results
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.
operm <- list()
for(i in c(1:50)){
  operm[[i]] <- get(load(paste0("output/permu/m2_Novelty_residuals_RankNormal_datarelease_12182019_", colnames(pheno)[14], "_", i, ".RData")))
  operm[[i]] <- operm[[i]]$OFA.total.distance.traveled
}
operm <- matrix(do.call(rbind, operm[c(1:50)]),ncol = 1)
class(operm) <- c("scan1perm", "matrix")
cutoff <- summary(operm, c(0.05, 0.1))[,1]

#genome-wide results
pdf(file = paste0("output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019", "_genomescan.pdf"), width = 14)
for(i in names(condi.m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(condi.m2.qtl.out[[i]]) # overall maximum LOD score
  plot(condi.m2.qtl.out[[i]], gm_DO2816_qc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
  abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
}
dev.off()

for(i in names(condi.m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(condi.m2.qtl.out[[i]]) # overall maximum LOD score
  plot(condi.m2.qtl.out[[i]], gm_DO2816_qc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  abline(h=cutoff[1], col="red",lty=2) # 0.05 cutoff
  abline(h=cutoff[2], col="red",lty=1) # 0.1 cutoff
}

#load blup results
x <- list()
#colnames(pheno)[14:41]
for(i in colnames(pheno)[14:41]){
  y <- get(load(paste0("output/blup/conditional.m2.blup_Novelty_residuals_RankNormal_datarelease_12182019_",i,".RData")))
  x[[i]] <- y[[i]]
}
condi.m2.qtl.blup = x

#coeffects plot
condi.m2.qtl.peak <- list()
pdf(file = paste0("output/conditional.m2.Novelty_residuals_RankNormal_datarelease_12182019", "_coeffgeneplot.pdf"), width = 14)
#names(condi.m2.qtl.out)
for(i in names(condi.m2.qtl.out)){
  #peak
  condi.m2.qtl.peak[[i]] <- find_peaks(condi.m2.qtl.out[[i]], gm_DO2816_qc$gmap,
                                       threshold=maxlod(condi.m2.qtl.out[[i]])-0.001,
                                       drop = 1.5)
  condi.m2.qtl.peak[[i]]$lodcolumn <- i
  print(condi.m2.qtl.peak[[i]])
  
  chr <- as.character(condi.m2.qtl.peak[[i]][condi.m2.qtl.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(condi.m2.coef[[i]],
               gm_DO2816_qc$gmap[chr],
               scan1_output=condi.m2.qtl.out[[i]],
               bgcolor="gray95", #legend="bottomleft",
               main =i)
  plot_coefCC(condi.m2.qtl.blup[[i]],
              gm_DO2816_qc$gmap[chr],
              scan1_output=condi.m2.qtl.out[[i]],
              bgcolor="gray95", #legend="bottomleft",
              main =i)
#   plot(condi.m2.snps[[i]]$lod,
#         condi.m2.snps[[i]]$snpinfo,
#         drop_hilit=1.5,
#         genes=condi.m2.genes[[i]],
#         main = i)
}
dev.off()
```
