---
title: "17_plot_qtl_mapping"
author: "Hao He"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This script will plot the qtl mapping results.

# library
```{r library, message=FALSE}
library(qtl2)
library(ggplot2)
load("data/Jackson_Lab_11_batches/gm_DO2816_qc.RData")
```

# plot scan result for m2 Prj01_RL-Acquisition_preqc_02142020.csv
```{r plot scan result for m2 Prj01_RL-Acquisition_preqc_02142020.csv,fig.height=14, fig.width=14, eval=TRUE}
pheno <- read.csv("data/pheno/preqc/Prj01_RL-Acquisition_preqc_02142020.csv", header = TRUE)
print("m2")
load("output/m2.Prj01_RL-Acquisition_preqc_02142020.RData")

#genome-wide results output/
pdf(file = paste0("output/m2.Prj01_RL-Acquisition_preqc_02142020", "_qtl2scan.pdf"), width = 14)
for(i in names(m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m2.qtl.out[[i]]) # overall maximum LOD score
  plot(m2.qtl.out[[i]], gm_DO2816_qc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  abline(h=summary(m2.permu[[i]], c(0.05, 0.1))[,1][1], col="red",lty=2) # 0.05 cutoff
  abline(h=summary(m2.permu[[i]], c(0.05, 0.1))[,1][2], col="red",lty=1) # 0.1 cutoff
  #effect plot
  chr <- as.character(m2.peak[[i]][m2.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
#  plot_coefCC(m2.coef[[i]], 
#              gm_DO2816_qc$gmap[chr], 
#              scan1_output=m2.qtl.out[[i]], 
#              bgcolor="gray95", #legend="bottomleft", 
#              main =i)
  plot_coefCC(m2.blup[[i]], 
              gm_DO2816_qc$gmap[chr], 
              scan1_output=m2.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
 # plot(m2.snps[[i]]$lod, 
 #      m2.snps[[i]]$snpinfo, 
 #      drop_hilit=1.5, 
 #      genes=m2.genes[[i]],
 #      main = i)
}
dev.off()

for(i in names(m2.qtl.out)){
  print(i)
  top_1 <- data.frame(pheno = i)
  top <- top_snps(m2.snps[[i]]$lod, m2.snps[[i]]$snpinfo)
  top <- cbind(top_1, top)
  write.table(top, file = "output/Prj01_RL-Acquisition_preqc_02142020-variantlist.csv", sep = ",",
            row.names = F, col.names = T, append = TRUE)

  if(dim(m2.genes[[i]]) == 0){
    m2.genes[[i]] = NA
  }
  
  topgene <- cbind(top_1, m2.genes[[i]])
  write.table(topgene, file = "output/Prj01_RL-Acquisition_preqc_02142020-genelist.csv", sep = ",",
              row.names = F, col.names = T, append = TRUE)
}
```

# plot scan result for m2 Prj01_RL-Reversal_preqc_02142020.csv
```{r plot scan result for m2 Prj01_RL-Reversal_preqc_02142020.csv,fig.height=14, fig.width=14, eval=TRUE}
pheno <- read.csv("data/pheno/preqc/Prj01_RL-Reversal_preqc_02142020.csv", header = TRUE)
print("m2")
load("output/m2.Prj01_RL-Reversal_preqc_02142020.RData")

#genome-wide results output/
pdf(file = paste0("output/m2.Prj01_RL-Reversal_preqc_02142020", "_qtl2scan.pdf"), width = 14)
for(i in names(m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m2.qtl.out[[i]]) # overall maximum LOD score
  plot(m2.qtl.out[[i]], gm_DO2816_qc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  abline(h=summary(m2.permu[[i]], c(0.05, 0.1))[,1][1], col="red",lty=2) # 0.05 cutoff
  abline(h=summary(m2.permu[[i]], c(0.05, 0.1))[,1][2], col="red",lty=1) # 0.1 cutoff
  #effect plot
  chr <- as.character(m2.peak[[i]][m2.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
#  plot_coefCC(m2.coef[[i]], 
#              gm_DO2816_qc$gmap[chr], 
#              scan1_output=m2.qtl.out[[i]], 
#              bgcolor="gray95", #legend="bottomleft", 
#              main =i)
  plot_coefCC(m2.blup[[i]], 
              gm_DO2816_qc$gmap[chr], 
              scan1_output=m2.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
 # plot(m2.snps[[i]]$lod, 
 #      m2.snps[[i]]$snpinfo, 
 #      drop_hilit=1.5, 
 #      genes=m2.genes[[i]],
 #      main = i)
}
dev.off()

for(i in names(m2.qtl.out)){
  print(i)
  top_1 <- data.frame(pheno = i)
  top <- top_snps(m2.snps[[i]]$lod, m2.snps[[i]]$snpinfo)
  top <- cbind(top_1, top)
  write.table(top, file = "output/Prj01_RL-Reversal_preqc_02142020-variantlist.csv", sep = ",",
            row.names = F, col.names = T, append = TRUE)

  if(dim(m2.genes[[i]]) == 0){
    m2.genes[[i]] = NA
  }
  
  topgene <- cbind(top_1, m2.genes[[i]])
  write.table(topgene, file = "output/Prj01_RL-Reversal_preqc_02142020-genelist.csv", sep = ",",
              row.names = F, col.names = T, append = TRUE)
}
```

# plot scan result for m2 Prj02_Sensitization_preqc_02142020.csv
```{r plot scan result for m2 Prj02_Sensitization_preqc_02142020.csv,fig.height=14, fig.width=14, eval=TRUE}
pheno <- read.csv("data/pheno/preqc/Prj02_Sensitization_preqc_02142020.csv", header = TRUE)
print("m2")
load("output/m2.Prj02_Sensitization_preqc_02142020.RData")

#genome-wide results output/
pdf(file = paste0("output/m2.Prj02_Sensitization_preqc_02142020", "_qtl2scan.pdf"), width = 14)
for(i in names(m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m2.qtl.out[[i]]) # overall maximum LOD score
  plot(m2.qtl.out[[i]], gm_DO2816_qc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  abline(h=summary(m2.permu[[i]], c(0.05, 0.1))[,1][1], col="red",lty=2) # 0.05 cutoff
  abline(h=summary(m2.permu[[i]], c(0.05, 0.1))[,1][2], col="red",lty=1) # 0.1 cutoff
  #effect plot
  chr <- as.character(m2.peak[[i]][m2.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
 # plot_coefCC(m2.coef[[i]], 
 #             gm_DO2816_qc$gmap[chr], 
 #             scan1_output=m2.qtl.out[[i]], 
 #             bgcolor="gray95", #legend="bottomleft", 
 #             main =i)
  plot_coefCC(m2.blup[[i]], 
              gm_DO2816_qc$gmap[chr], 
              scan1_output=m2.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
 # plot(m2.snps[[i]]$lod, 
 #      m2.snps[[i]]$snpinfo, 
 #      drop_hilit=1.5, 
 #      genes=m2.genes[[i]],
 #      main = i)
}
dev.off()

for(i in names(m2.qtl.out)){
  print(i)
  top_1 <- data.frame(pheno = i)
  top <- top_snps(m2.snps[[i]]$lod, m2.snps[[i]]$snpinfo)
  top <- cbind(top_1, top)
  write.table(top, file = "output/Prj02_Sensitization_preqc_02142020-variantlist.csv", sep = ",",
            row.names = F, col.names = T, append = TRUE)
  
  if(dim(m2.genes[[i]]) == 0){
    m2.genes[[i]] = NA
  }

  topgene <- cbind(top_1, m2.genes[[i]])
  write.table(topgene, file = "output/Prj02_Sensitization_preqc_02142020-genelist.csv", sep = ",",
              row.names = F, col.names = T, append = TRUE)
}

```

# plot scan result for m2 Prj04_IVSA_preqc_02142020.csv
```{r plot scan result for m2 Prj04_IVSA_preqc_02142020.csv,fig.height=14, fig.width=14, eval=TRUE}
pheno <- read.csv("data/pheno/preqc/Prj04_IVSA_preqc_02142020.csv", header = TRUE)
print("m2")
load("output/m2.Prj04_IVSA_preqc_02142020.RData")

#genome-wide results output/
pdf(file = paste0("output/m2.Prj04_IVSA_preqc_02142020", "_qtl2scan.pdf"), width = 14)
for(i in names(m2.qtl.out)){
  par(mar=c(5.1, 4.1, 1.1, 1.1))
  ymx <- maxlod(m2.qtl.out[[i]]) # overall maximum LOD score
  plot(m2.qtl.out[[i]], gm_DO2816_qc$gmap, lodcolumn=1,
       col="slateblue", ylim=c(0, ymx*1.02),
       main=i)
  abline(h=summary(m2.permu[[i]], c(0.05, 0.1))[,1][1], col="red",lty=2) # 0.05 cutoff
  abline(h=summary(m2.permu[[i]], c(0.05, 0.1))[,1][2], col="red",lty=1) # 0.1 cutoff
  #effect plot
  chr <- as.character(m2.peak[[i]][m2.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  #plot_coefCC(m2.coef[[i]], 
  #            gm_DO2816_qc$gmap[chr], 
  #            scan1_output=m2.qtl.out[[i]], 
  #            bgcolor="gray95", #legend="bottomleft", 
  #            main =i)
  plot_coefCC(m2.blup[[i]], 
              gm_DO2816_qc$gmap[chr], 
              scan1_output=m2.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =i)
 # plot(m2.snps[[i]]$lod, 
 #      m2.snps[[i]]$snpinfo, 
 #      drop_hilit=1.5, 
 #      genes=m2.genes[[i]],
 #      main = i)
}
dev.off()

for(i in names(m2.qtl.out)){
  print(i)
  top_1 <- data.frame(pheno = i)
  top <- top_snps(m2.snps[[i]]$lod, m2.snps[[i]]$snpinfo)
  top <- cbind(top_1, top)
  write.table(top, file = "output/Prj04_IVSA_preqc_02142020-variantlist.csv", sep = ",",
            row.names = F, col.names = T, append = TRUE)

  if(dim(m2.genes[[i]]) == 0){
    m2.genes[[i]] = NA
  }
  
  topgene <- cbind(top_1, m2.genes[[i]])
  write.table(topgene, file = "output/Prj04_IVSA_preqc_02142020-genelist.csv", sep = ",",
              row.names = F, col.names = T, append = TRUE)
}
```
